surfactant/infoextractors/file_decompression.py

```diff
@@ -46,21 +46,40 @@ def extract_file_info(
     current_context: Optional[ContextEntry],
 ) -> Optional[Dict[str, Any]]:
     # Check if the file is compressed and get its format
+
     compression_format = supports_file(filetype)
     if not compression_format:
         return None
 
-    # Decompress the file based on its format
-    temp_folder = check_compression_type(filename, compression_format)
-
-    # Add a new ContextEntry for the temp dir
+    install_prefix = ""
+    extract_paths = []
+
+    # Check that archive key exists and filename is same as archive file
+    if current_context.archive and current_context.archive == filename:
+        # If archive has empty list of extractPaths, an entry to the queue will be added
+        if current_context.extractPaths == []:
+            temp_folder = check_compression_type(current_context.archive, compression_format)
+            install_prefix = current_context.installPrefix
+            extract_paths = [temp_folder]
+            logger.info(f"New ContextEntry added for archive: {current_context.archive}")
+        # Assume archive is already extracted if the extractPaths is not empty
+        else:
+            logger.info(
+                f"Already extracted, skipping extraction for archive: {current_context.archive}"
+            )
+            return None
+    # Assume there is a compressed file in the extractPaths
+    elif current_context.installPrefix == "/":
+        # Decompress the file based on its format
+        temp_folder = check_compression_type(filename, compression_format)
+        extract_paths = [temp_folder]
+        logger.info(f"New ContextEntry added for extracted files: {temp_folder}")
```

Comment:
```suggestion

    # Decompress the file based on its format
    temp_folder = check_compression_type(filename, compression_format)
    extract_paths = [temp_folder]
    logger.info(f"New ContextEntry added for extracted files: {temp_folder}")
```

Only a single call to decompress a file is needed, there shouldn't be anything special to this part that makes decompressing a file listed as an "archive" different than one found within the extract paths folder that surfactant is walking through.